# Market basket analysis of data in list form

Q) Find the items with at least 60% support.

Which three itemset is a sure shot indicator that they will maximise profits if kept together

Dataset (Data is in the form of Python list)

In [57]:

import numpy as np
import pandas as pd
from apyori import apriori
dataset = [["Shampoo", 'Conditioner',' Soap', 'Shower Gel', 'Facewash',
'Shaving Foam'],
['Body Lotion','Soap','Aftershave', 'Facewash', 'Shower Gel', 'Shaving Foam'],
['Shampoo', 'Soap', 'Shower Gel', 'Facewash'],
['Conditioner', 'Shaving Foam', 'Soap', 'Aftershave', 'Shower Gel'],
['Soap', 'Shampoo', 'Aftershave', 'Shower Gel','Facewash'],
['Shower Gel', 'Soap', 'Facewash', 'Body Lotion'],
['Body Lotion', 'Shampoo', 'Facewash'],
['Body Lotion', 'Aftershave', 'Soap', 'Shower Gel', 'Facewash']]
dataset

[['Shampoo', 'Conditioner', ' Soap', 'Shower Gel', 'Facewash', 'Shaving Foam'],
 ['Body Lotion',
  'Soap',
  'Aftershave',
  'Facewash',
  'Shower Gel',
  'Shaving Foam'],
 ['Shampoo', 'Soap', 'Shower Gel', 'Facewash'],
 ['Conditioner', 'Shaving Foam', 'Soap', 'Aftershave', 'Shower Gel'],
 ['Soap', 'Shampoo', 'Aftershave', 'Shower Gel', 'Facewash'],
 ['Shower Gel', 'Soap', 'Facewash', 'Body Lotion'],
 ['Body Lotion', 'Shampoo', 'Facewash'],
 ['Body Lotion', 'Aftershave', 'Soap', 'Shower Gel', 'Facewash']]

In [58]:
d = list(apriori(dataset))
df = pd.DataFrame(d)
df

items  support  \
0                                              ( Soap)    0.125   
1                                         (Aftershave)    0.500   
2                                        (Body Lotion)    0.500   
3                                        (Conditioner)    0.250   
4                                           (Facewash)    0.875   
..                                                 ...      ...   
140  (Shower Gel, Soap, Aftershave, Facewash, Shavi...    0.125   
141  (Shower Gel, Body Lotion, Soap, Facewash, Shav...    0.125   
142  (Shampoo, Shower Gel, Conditioner, Facewash, S...    0.125   
143  (Shampoo,  Soap, Shower Gel, Conditioner, Face...    0.125   
144  (Shower Gel, Body Lotion, Soap, Aftershave, Fa...    0.125   

                                    ordered_statistics  
0                          [((), ( Soap), 0.125, 1.0)]  
1                       [((), (Aftershave), 0.5, 1.0)]  
2                      [((), (Body Lotion), 0.5, 1.0)]  
3                     [((), (Conditioner), 0.25, 1.0)]  
4                       [((), (Facewash), 0.875, 1.0)]  
..                                                 ...  
140  [((), (Shower Gel, Soap, Aftershave, Facewash,...  
141  [((), (Shower Gel, Soap, Body Lotion, Facewash...  
142  [((), (Shampoo, Shaving Foam, Shower Gel, Face...  
143  [((), (Shampoo, Shaving Foam,  Soap, Shower Ge...  
144  [((), (Shower Gel, Soap, Body Lotion, Facewash...  

[145 rows x 3 columns]

In [59]:
df_sort = df.loc[df['support']>=.6]
df_sort = df_sort.reset_index(drop=True)
df_sort
# index 6 shows support higher than 60 percent 

items  support  \
0                    (Facewash)    0.875   
1                  (Shower Gel)    0.875   
2                        (Soap)    0.750   
3        (Facewash, Shower Gel)    0.750   
4              (Facewash, Soap)    0.625   
5            (Soap, Shower Gel)    0.750   
6  (Soap, Facewash, Shower Gel)    0.625   

                                  ordered_statistics  
0                     [((), (Facewash), 0.875, 1.0)]  
1                   [((), (Shower Gel), 0.875, 1.0)]  
2                          [((), (Soap), 0.75, 1.0)]  
3  [((), (Facewash, Shower Gel), 0.75, 1.0), ((Fa...  
4  [((), (Facewash, Soap), 0.625, 1.0), ((Facewas...  
5  [((), (Shower Gel, Soap), 0.75, 1.0), ((Shower...  
6  [((), (Shower Gel, Facewash, Soap), 0.625, 1.0...

# Above data frame Shows set of single , 2 items and 3 items which have support greater than 60 percent. 

**Soap, Facewash, Shower Gel has support of 62.5%** and if placed together it could boost profit of the retial store/outlet
